In [1]:
import pandas as pd


In [2]:
df_transactions = pd.read_csv('../data/raw/train_transaction.csv')
df_identity = pd.read_csv('../data/raw/train_identity.csv')

In [5]:
df_transactions.head(10)

TransactionID  isFraud  TransactionDT  TransactionAmt ProductCD  card1  \
0        2987000        0          86400            68.5         W  13926   
1        2987001        0          86401            29.0         W   2755   
2        2987002        0          86469            59.0         W   4663   
3        2987003        0          86499            50.0         W  18132   
4        2987004        0          86506            50.0         H   4497   
5        2987005        0          86510            49.0         W   5937   
6        2987006        0          86522           159.0         W  12308   
7        2987007        0          86529           422.5         W  12695   
8        2987008        0          86535            15.0         H   2803   
9        2987009        0          86536           117.0         W  17399   

   card2  card3       card4  card5  ... V330  V331  V332  V333  V334 V335  \
0    NaN  150.0    discover  142.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
1  404.0  150.0  mastercard  102.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
2  490.0  150.0        visa  166.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
3  567.0  150.0  mastercard  117.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
4  514.0  150.0  mastercard  102.0  ...  0.0   0.0   0.0   0.0   0.0  0.0   
5  555.0  150.0        visa  226.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
6  360.0  150.0        visa  166.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
7  490.0  150.0        visa  226.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   
8  100.0  150.0        visa  226.0  ...  0.0   0.0   0.0   0.0   0.0  0.0   
9  111.0  150.0  mastercard  224.0  ...  NaN   NaN   NaN   NaN   NaN  NaN   

  V336  V337  V338  V339  
0  NaN   NaN   NaN   NaN  
1  NaN   NaN   NaN   NaN  
2  NaN   NaN   NaN   NaN  
3  NaN   NaN   NaN   NaN  
4  0.0   0.0   0.0   0.0  
5  NaN   NaN   NaN   NaN  
6  NaN   NaN   NaN   NaN  
7  NaN   NaN   NaN   NaN  
8  0.0   0.0   0.0   0.0  
9  NaN   NaN   NaN   NaN  

[10 rows x 394 columns]

In [6]:
df_identity.head()

TransactionID  id_01     id_02  id_03  id_04  id_05  id_06  id_07  id_08  \
0        2987004    0.0   70787.0    NaN    NaN    NaN    NaN    NaN    NaN   
1        2987008   -5.0   98945.0    NaN    NaN    0.0   -5.0    NaN    NaN   
2        2987010   -5.0  191631.0    0.0    0.0    0.0    0.0    NaN    NaN   
3        2987011   -5.0  221832.0    NaN    NaN    0.0   -6.0    NaN    NaN   
4        2987016    0.0    7460.0    0.0    0.0    1.0    0.0    NaN    NaN   

   id_09  ...                id_31  id_32      id_33           id_34  id_35  \
0    NaN  ...  samsung browser 6.2   32.0  2220x1080  match_status:2      T   
1    NaN  ...   mobile safari 11.0   32.0   1334x750  match_status:1      T   
2    0.0  ...          chrome 62.0    NaN        NaN             NaN      F   
3    NaN  ...          chrome 62.0    NaN        NaN             NaN      F   
4    0.0  ...          chrome 62.0   24.0   1280x800  match_status:2      T   

  id_36 id_37  id_38  DeviceType                     DeviceInfo  
0     F     T      T      mobile  SAMSUNG SM-G892A Build/NRD90M  
1     F     F      T      mobile                     iOS Device  
2     F     T      T     desktop                        Windows  
3     F     T      T     desktop                            NaN  
4     F     T      T     desktop                          MacOS  

[5 rows x 41 columns]

In [6]:
df = df_transactions.merge(df_identity,on='TransactionID', how='left')

In [7]:
df.head(10)

TransactionID  isFraud  TransactionDT  TransactionAmt ProductCD  card1  \
0        2987000        0          86400            68.5         W  13926   
1        2987001        0          86401            29.0         W   2755   
2        2987002        0          86469            59.0         W   4663   
3        2987003        0          86499            50.0         W  18132   
4        2987004        0          86506            50.0         H   4497   
5        2987005        0          86510            49.0         W   5937   
6        2987006        0          86522           159.0         W  12308   
7        2987007        0          86529           422.5         W  12695   
8        2987008        0          86535            15.0         H   2803   
9        2987009        0          86536           117.0         W  17399   

   card2  card3       card4  card5  ...                id_31  id_32  \
0    NaN  150.0    discover  142.0  ...                  NaN    NaN   
1  404.0  150.0  mastercard  102.0  ...                  NaN    NaN   
2  490.0  150.0        visa  166.0  ...                  NaN    NaN   
3  567.0  150.0  mastercard  117.0  ...                  NaN    NaN   
4  514.0  150.0  mastercard  102.0  ...  samsung browser 6.2   32.0   
5  555.0  150.0        visa  226.0  ...                  NaN    NaN   
6  360.0  150.0        visa  166.0  ...                  NaN    NaN   
7  490.0  150.0        visa  226.0  ...                  NaN    NaN   
8  100.0  150.0        visa  226.0  ...   mobile safari 11.0   32.0   
9  111.0  150.0  mastercard  224.0  ...                  NaN    NaN   

       id_33           id_34  id_35 id_36 id_37  id_38  DeviceType  \
0        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
1        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
2        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
3        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
4  2220x1080  match_status:2      T     F     T      T      mobile   
5        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
6        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
7        NaN             NaN    NaN   NaN   NaN    NaN         NaN   
8   1334x750  match_status:1      T     F     F      T      mobile   
9        NaN             NaN    NaN   NaN   NaN    NaN         NaN   

                      DeviceInfo  
0                            NaN  
1                            NaN  
2                            NaN  
3                            NaN  
4  SAMSUNG SM-G892A Build/NRD90M  
5                            NaN  
6                            NaN  
7                            NaN  
8                     iOS Device  
9                            NaN  

[10 rows x 434 columns]

In [8]:
df.shape #nombre de lignes x colonnes

(590540, 434)

In [9]:
df.dtypes.head(20)

TransactionID       int64
isFraud             int64
TransactionDT       int64
TransactionAmt    float64
ProductCD          object
card1               int64
card2             float64
card3             float64
card4              object
card5             float64
card6              object
addr1             float64
addr2             float64
dist1             float64
dist2             float64
P_emaildomain      object
R_emaildomain      object
C1                float64
C2                float64
C3                float64
dtype: object

In [10]:
#pourcentage de valeurs manquantes pour chaque colonne
df.isna().mean().sort_values(ascending=False) 

id_24            0.991962
id_25            0.991310
id_07            0.991271
id_08            0.991271
id_21            0.991264
                   ...   
C11              0.000000
C14              0.000000
C13              0.000000
C12              0.000000
TransactionID    0.000000
Length: 434, dtype: float64

In [11]:
#proportion de valeurs 0 (transaction normale) et 1 (fraude)
df['isFraud'].value_counts(normalize=True) 

isFraud
0    0.96501
1    0.03499
Name: proportion, dtype: float64

In [12]:
#Repérer les 5 features numériques les plus corrélées avec isFraud
df.corr(numeric_only=True)['isFraud'].abs().sort_values(ascending=False).head(6)

isFraud    1.000000
V257       0.383060
V246       0.366878
V244       0.364129
V242       0.360590
V201       0.328005
Name: isFraud, dtype: float64

### 📋 Résumé – EDA Initiale IEEE-CIS Fraud Detection
#### 🔍 Dimensions du dataset :

590 540 lignes

434 colonnes après fusion train_transaction + train_identity

#### 🔍 Proportion de fraudes (isFraud) :

Fraudes (isFraud = 1) : 3.5 %

Transactions normales (isFraud = 0) : 96.5 %

#### 🔍 Colonnes avec plus de 80 % de valeurs manquantes :

Exemple : id_24, id_25, id_07, id_08, id_21

#### 🔍 Top 5 features numériques corrélées avec isFraud :
V257

V246

V244

V242

V201

(Corrélations modérées (~0.35), indiquant un signal utile pour prédire la fraude.)
